openedx_learning/apps/authoring/collections/api.py

```diff
@@ -70,6 +87,90 @@ def update_collection(
     return collection
 
 
+def add_to_collections(
+    collections_qset: QuerySet[Collection],
+    contents_qset: QuerySet[PublishableEntity],
+) -> int:
+    """
+    Adds a QuerySet of PublishableEntities to a QuerySet of Collections.
+
+    Records are created in bulk, and so integrity errors are deliberately ignored: they indicate that the content(s)
+    have already been added to the collection(s).
+
+    Returns the number of entities added (including any that already exist).
+    """
+    collection_objects = []
+    object_ids = contents_qset.values_list("pk", flat=True)
+    collection_ids = collections_qset.values_list("pk", flat=True)
+
+    for collection_id in collection_ids:
+        for object_id in object_ids:
+            collection_objects.append(
+                CollectionObject(
+                    collection_id=collection_id,
+                    object_id=object_id,
+                )
+            )
+
+    created = []
+    if collection_objects:
+        created = CollectionObject.objects.bulk_create(
+            collection_objects,
+            ignore_conflicts=True,
+        )
+
+        # Update the modified date for all the provided Collections.
+        # (Ignoring conflicts means we don't know which ones were modified.)
+        collections_qset.update(modified=timezone.now())
```

Comment:
`timezone.now()` doesn't accept a `tz` parameter, but the [Django docs](https://docs.djangoproject.com/en/4.2/ref/utils/#django.utils.timezone.now) say:

> If [USE_TZ](https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-USE_TZ) is True, this will be an [aware](https://docs.djangoproject.com/en/4.2/topics/i18n/timezones/#naive-vs-aware-datetimes) datetime representing the current time in UTC. Note that [now()](https://docs.djangoproject.com/en/4.2/ref/utils/#django.utils.timezone.now) will always return times in UTC regardless of the value of [TIME_ZONE](https://docs.djangoproject.com/en/4.2/ref/settings/#std-setting-TIME_ZONE);

If someone has un-set `settings.USE_TZ`, then they'll probably have all kinds of problems, so I think this is ok as-is.

Reply:
Other parts of the code use `datetime.now(tz=timezone.utc)` - so I think you can just use that.

Reply:
@pomegranited: I think that you're right that it will probably break a bunch of things if they don't have `USE_TZ=True`. I hadn't realized that it would always return UTC timezones if that setting was on–I had assumed it would return whatever the configured `TIME_ZONE` setting was set to.

As @bradenmacdonald notes, I've always just used `datetime.now(tz=timezone.utc)`.

So I have a mild preference for `datetime.now(tz=timezone.utc)`, but I'm fine with either.